frontend/app/login/components/register-form.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+
+const formSchema = z.object({
+  firstName: z.string().min(2).max(20),
+  lastName: z.string().min(2).max(20),
+  email: z.string().email("Invalid email"),
+  username: z.string().min(2).max(20),
+  password: z.string().min(8),
+});
+
+export function RegisterForm() {
+  // 1. Define your form.
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      firstName: "",
+      lastName: "",
+      email: "",
+      username: "",
+      password: "",
+    },
```

Comment:
Here as well, can't have default values (will not pass anyway since they have a minimum chars allowed) and custom err msgs.

Reply:
Ok, I will remove the default values....